app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,44 @@
+class QuestionsController < ApplicationController
+    before_action :authenticate_user!, only: [:new, :create,:idnex,:show]
+
+    def show
+        @question = Question.find(params[:id])
+        @answer = Answer.new
+        @answers = Answer.all
```

Comment:
質問に対応する回答を取らねばなりません。他の質問の回答を表示してはいけません。

```suggestion
        @answers = @question.answers.order(id: :desc)
        @answer = Answer.new
```

Reply:
表示する回答がどの質問に対応しているかの関連付けを、上で定義した@questionを用いて行う必要があるのですね。